crates/common/src/local_db/decode.rs

```diff
@@ -215,9 +216,56 @@ fn decode_event<E: SolEvent>(event: &LogEntryResponse) -> Result<E, DecodeError>
     E::decode_raw_log(topics, &data).map_err(|err| DecodeError::AbiDecode(err.to_string()))
 }
 
+pub fn sort_decoded_events_by_block_and_log(
+    events: &mut [DecodedEventData<DecodedEvent>],
+) -> Result<(), LocalDbError> {
+    let mut keyed = Vec::with_capacity(events.len());
+    for (idx, event) in events.iter().enumerate() {
+        let block = parse_u64_hex_or_dec(&event.block_number).map_err(|err| {
+            LocalDbError::InvalidBlockNumberString {
+                value: event.block_number.clone(),
+                source: err,
+            }
+        })?;
+        let log_index = parse_u64_hex_or_dec(&event.log_index).map_err(|err| {
+            LocalDbError::InvalidLogIndex {
+                value: event.log_index.clone(),
+                source: err,
+            }
+        })?;
+        keyed.push((idx, block, log_index));
+    }
+
+    keyed.sort_by(|a, b| {
+        a.1.cmp(&b.1)
+            .then_with(|| a.2.cmp(&b.2))
+            .then_with(|| a.0.cmp(&b.0))
+    });
+
+    let original = events.to_vec();
+    for (position, (idx, _, _)) in keyed.into_iter().enumerate() {
+        events[position] = original[idx].clone();
+    }
+
+    Ok(())
+}
+
+fn parse_u64_hex_or_dec(value: &str) -> Result<u64, std::num::ParseIntError> {
```

Comment:
I feel like I've seen this before - as a rule I think it would be good to put generic helpers like this in common/utils